python/sglang/srt/grpc_server.py

```diff
@@ -0,0 +1,74 @@
+import traceback
+from typing import Any, AsyncGenerator, Callable, Dict
+
+import grpc
+
+from sglang.srt.managers.io_struct import GenerateReqInput
+from sglang.srt.proto import completion_pb2, completion_pb2_grpc
+
+
+class CompletionServicer(completion_pb2_grpc.CompletionServiceServicer):
+    def __init__(
+        self,
+        generate_request: Callable[
+            [GenerateReqInput], AsyncGenerator[Dict[str, Any], None]
+        ],
+    ):
+        self.generate_request = generate_request
+
+    async def Complete(
+        self,
+        request: completion_pb2.CompletionRequest,
+        context: grpc.aio.ServicerContext,
+    ) -> AsyncGenerator[completion_pb2.CompletionResponse, None]:
+        try:
+            # Convert gRPC request to internal format
+            adapted_request = GenerateReqInput(
+                text=request.prompt,
+                sampling_params={
+                    "max_new_tokens": request.max_tokens,
+                    "temperature": request.temperature,
+                    "top_p": request.top_p,
```

Comment:
this part is dangerous. grpc protobuf has a default value 0 for unset values, and it means this top_p, top_k could be different from the default (1 and -1 respectively)

I think we need some sort of mechanism to set the default value? one potential solution is to use optional field in protobuf and set the correct default value when they are not provided